tests/test_selected_rows.py

```diff
@@ -95,3 +97,75 @@ def selected(n):
     dash_duo.wait_for_text_to_equal(
         "#selections-single-output", "You selected athlete: Natalie Coughlin"
     )
+
+def test_sr2_selected_rows_rowdata(dash_duo):
```

Comment:
Reviewing the test cases in #274:

We're covering:
- Cannot output rowData and selectedRows in the same callback

But confirming this covers:
- Inconsistent data type when selectedRows is an input depending how selectedRows was specified
- When using rowIds and checkbox selection, the checkboxes are never visibly selected

Could you add a comment to this test describing these cases as well so it's clear what it's testing for?

This example is much clearer and more concise overall than the originals, kudos!